test/RibbonThetaVault.ts

```diff
@@ -3138,6 +3138,112 @@ function behavesLikeRibbonOptionsVault(params: {
       });
     });
 
+    describe("#startAuction", () => {
+      let otoken: Contract;
+      let initialOtokenBalance: BigNumber;
+      let startOtokenPrice: BigNumber;
+      const depositAmount = params.depositAmount;
+
+      time.revertToSnapshotAfterEach(async function () {
+        await depositIntoVault(params.collateralAsset, vault, depositAmount);
+
+        await setupOracle(
+          params.asset,
+          params.chainlinkPricer,
+          ownerSigner,
+          params.protocol
+        );
+
+        await vault.connect(ownerSigner).commitAndClose();
+        startOtokenPrice = await vault.currentOtokenPremium();
+
+        await time.increaseTo((await vault.nextOptionReadyAt()).toNumber() + 1);
+
+        await vault.connect(keeperSigner).rollToNextOption();
+        const { currentOption } = await vault.optionState();
+        otoken = await ethers.getContractAt("IERC20", currentOption);
+        initialOtokenBalance = await otoken.balanceOf(gnosisAuction.address);
+      });
+
+      it("restarts the auction process", async () => {
+        await time.increaseTo(
+          (await provider.getBlock("latest")).timestamp + auctionDuration
+        );
+
+        // we simulate settling the auction without any bids
+        await gnosisAuction
+          .connect(userSigner)
+          .settleAuction(await gnosisAuction.auctionCounter());
+
+        const afterOtokenBalance = await otoken.balanceOf(vault.address);
+        assert.bnEqual(initialOtokenBalance, afterOtokenBalance);
+
+        // We increase the discount so the otoken min price should go down
+        await vault
+          .connect(keeperSigner)
+          .setPremiumDiscount(BigNumber.from("800"));
+
+        await vault.connect(keeperSigner).startAuction();
+
+        assert.bnEqual(
+          await otoken.balanceOf(gnosisAuction.address),
+          initialOtokenBalance
+        );
+
+        // otoken price is decreased on the auction
```

Comment:
is it possible it is less than bc price of underlying changed (so relative to same strike it could be cheaper). maybe make sure it is less than by the premium discount at least

Reply:
Don't really want to add another restriction so we have more flexibility here, for whatever reasons the auction doesn't fill. eg maybe there just aren't bidders so we need to readjust the price higher or lower.